servers/rendering/rasterizer_rd/shaders/scene_high_end.glsl

```diff
@@ -2376,6 +2497,93 @@ FRAGMENT_SHADER_CODE
 
 #ifdef MODE_RENDER_DEPTH
 
+#ifdef MODE_RENDER_SDF
+
+	{
+		vec3 local_pos = (scene_data.sdf_to_bounds * vec4(vertex, 1.0)).xyz;
+		ivec3 grid_pos = scene_data.sdf_offset + ivec3(local_pos * vec3(scene_data.sdf_size));
+
+		uint albedo16 = 0x1; //solid flag
+		albedo16 |= clamp(uint(albedo.r * 31.0), 0, 31) << 11;
+		albedo16 |= clamp(uint(albedo.g * 31.0), 0, 31) << 6;
+		albedo16 |= clamp(uint(albedo.b * 31.0), 0, 31) << 1;
+
+		imageStore(albedo_volume_grid, grid_pos, uvec4(albedo16));
+
+		uint facing_bits = 0;
+		const vec3 aniso_dir[6] = vec3[](
+				vec3(1, 0, 0),
+				vec3(0, 1, 0),
+				vec3(0, 0, 1),
+				vec3(-1, 0, 0),
+				vec3(0, -1, 0),
+				vec3(0, 0, -1));
+
+		vec3 cam_normal = mat3(scene_data.camera_matrix) * normal;
+
+		for (uint i = 0; i < 6; i++) {
+			if (dot(cam_normal, aniso_dir[i]) > 0.001) {
+				facing_bits |= (1 << i);
+			}
+		}
+
+		imageAtomicOr(geom_facing_grid, grid_pos, facing_bits); //store facing bits
```

Comment:
This line fails on macOS with following error:

```
[mvk-error] VK_ERROR_INITIALIZATION_FAILED: Shader library compile failed (Error code 3):
Compilation failed:

program_source:390:87: error: array subscript is not an integer
    uint _331 = atomic_fetch_or_explicit((device atomic_uint*)&geom_facing_grid_atomic[grid_pos], facing_bits, memory_order_relaxed);
                                                                                      ^~~~~~~~~
```

Apparently, Metal doesn't support atomic ops on texture storage in fragment shaders at all (https://github.com/KhronosGroup/SPIRV-Cross/issues/835).